src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/CombineAttributeArrays.cpp

```diff
@@ -0,0 +1,210 @@
+#include "CombineAttributeArrays.hpp"
+
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataGroup.hpp"
+
+using namespace complex;
+
+/**
+ * @brief The CombineAttributeArraysImpl class is a templated private implementation that deals with
+ * combining the various input arrays into one contiguous array
+ */
+template <typename DataType>
+class CombineAttributeArraysImpl
+{
+private:
+  const CombineAttributeArraysInputValues* m_InputValues = nullptr;
+  const std::vector<DataObject*> m_InputArrays;
+  DataObject* m_OutputArray = nullptr;
+
+public:
+  typedef DataArray<DataType> DataArrayType;
+
+  CombineAttributeArraysImpl(const CombineAttributeArraysInputValues* inputValues, std::vector<DataObject*>& inputArrays, DataObject* outputArray)
+  : m_InputValues(inputValues)
+  , m_InputArrays(inputArrays)
+  , m_OutputArray(outputArray)
+  {
+  }
+
+  virtual ~CombineAttributeArraysImpl() = default;
+
+  // -----------------------------------------------------------------------------
+  Result<> operator()()
+  {
+    using OutputArrayType = DataArray<DataType>;
+    using InputArrayType = DataArray<DataType>;
+
+    auto& outputArray = dynamic_cast<OutputArrayType&>(*m_OutputArray);
+    int32_t numArrays = m_InputArrays.size();
+
+    std::vector<InputArrayType*> inputArrays;
+
+    for(size_t i = 0; i < numArrays; i++)
+    {
+      inputArrays.push_back(dynamic_cast<InputArrayType*>(m_InputArrays[i]));
+    }
+
+    size_t numTuples = inputArrays[0]->getNumberOfTuples();
```

Comment:
@imikejackson Okay so I know that the filter makes sure that we don't have an empty input array but since this class could technically be used without the corresponding filter we should probably do a check to make sure `inputArrays` isn't empty before accessing the data in it.

Reply:
@jmarquisbq For the Algorithms class, no other algorithm class really does this though. Whether that is by design or we didn't think about it. The implicit understanding of using an Algorithm class from somewhere else is that the developer has already done those checks ahead of time and the variables being passed into the algorithm class have passed some sort of sanity check.
I can put in a few checks but they feel awfully redundant....

Reply:
```    if(numArrays == 0)
    {
      return MakeWarningVoidResult(1, "No arrays were selected to combine.");
    }
```
it is a simple enough check...

Reply:
@imikejackson Yeah I realize that it's redundant in this context since the filter already checks that so if you are okay with forcing the user to always put those checks in the filter then I'm fine with this the way it is but maybe put a comment or something so it's easier for the user to pick that up if they try to reuse this elsewhere?